torch/csrc/autograd/FunctionsManual.cpp

```diff
@@ -3661,37 +3661,38 @@ Tensor linalg_lstsq_jvp(
 }
 
 std::tuple<Tensor, Tensor> linalg_lstsq_backward(
-    const Tensor& grad,
+    const Tensor& gX_,
     const Tensor& A,
-    const Tensor& B,
-    const c10::optional<double> rcond,
-    const c10::optional<c10::string_view> driver,
-    const std::array<bool, 2>& grad_input_mask) {
+    const Tensor& B_) {
   at::NoTF32Guard disable_tf32;
-  Tensor A_grad, B_grad;
-  if (!grad.defined()) {
-    return std::make_tuple(A_grad, B_grad);
+  auto A_requires_grad = A.requires_grad();
```

Comment:
You should use `grad_input_mask` that is provided to the formula to decide what to do. Not the requires_grad field.

Reply:
This is important for cases where user uses `autograd.grad()` or `backward(inputs=)`: some Tensors might require gradients but their gradient is not needed for this particular call to backward.

Reply:
ahhh, fair. We should fix that in many other linalg formulas then!